packages/forms/backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/components/LabelWidgetComponent.java

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.forms.components;
+
+import java.util.List;
+import java.util.Objects;
+
+import org.eclipse.sirius.components.forms.description.LabelDescription;
+import org.eclipse.sirius.components.forms.elements.LabelWidgetElementProps;
+import org.eclipse.sirius.components.forms.elements.LabelWidgetElementProps.Builder;
+import org.eclipse.sirius.components.forms.validation.DiagnosticComponent;
+import org.eclipse.sirius.components.forms.validation.DiagnosticComponentProps;
+import org.eclipse.sirius.components.representations.Element;
+import org.eclipse.sirius.components.representations.IComponent;
+import org.eclipse.sirius.components.representations.VariableManager;
+
+/**
+ * The component used to render the label widget.
+ *
+ * @author fbarbin
+ */
+public class LabelWidgetComponent implements IComponent {
+
+    private LabelWidgetComponentProps props;
+
+    public LabelWidgetComponent(LabelWidgetComponentProps props) {
+        this.props = Objects.requireNonNull(props);
+    }
+
+    @Override
+    public Element render() {
+        VariableManager variableManager = this.props.getVariableManager();
+        LabelDescription labelDescription = this.props.getLabelDescription();
+
+        String id = labelDescription.getIdProvider().apply(variableManager);
+        String label = labelDescription.getLabelProvider().apply(variableManager);
+        String value = labelDescription.getValueProvider().apply(variableManager);
+        variableManager.put(LabelDescription.VALUE, value);
+        String display = labelDescription.getDisplayProvider().apply(variableManager);
```

Comment:
I really don't understand why you would have both a `value` and a `display`, it does not make sense. Just ask the specifier to compute a string and that's it. There no need to compute a `value` string to give it back to the specifier to compute a `display` string.

Reply:
In Eclipse EEF, the label widget has both a `valueExpression` and a `displayExpression` but there was a specific reason for that. In the very first version of the new Eclipse EEF, the label widget did not exist and it was instead the reference widget in a ready only mode (only the label was visible). After that, we decided to had the label widget as a dedicated widget.

One day, we changed that to offer a better support of the reference widget and we choose to separate the label appearance from the reference widget and instead migrate existing versions of the reference widget to the new label widget.

The reference widget had both a `valueExpression` and a `displayExpression` (which can use the `value` variable) and we needed to replicate this mechanism to ensure a smooth migration. The `LabelDescription#displayExpression` only exists in Eclipse EEF because we had to migrate automatically old instances of `eef.ecore` which were already being used by customers. This was also the reason for the introduction of widget actions on label.

See the first two comments on this [issue](https://bugs.eclipse.org/bugs/show_bug.cgi?id=500472). All that can also be applied to the hyperlink widget of Eclipse EEF (`valueExpression` + `displayExpression` + widget actions).